iothub/device/src/Common/Security/CertificateInstaller.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.Azure.Devices.Shared;
+
+namespace Microsoft.Azure.Devices.Client
+{
+    internal static class CertificateInstaller
+    {
+        private static readonly object s_certOperationsLock = new object();
+
+        /// <summary>
+        /// Ensures the specified certs (presumably in a chain) are installed in the cert store.
+        /// </summary>
+        /// <remarks>
+        /// Because Intermediate Authorities may have been issued by the uploaded CA, the application must present the full chain of
+        /// certificates from the one used during authentication to the one uploaded to the service.
+        /// See <see href="TODO (Sample ref) - Sindhu"/>
+        /// for more information.
+        /// </remarks>
+        /// <param name="certificates">The certificate chain to ensure is installed.</param>
+        public static void EnsureChainIsInstalled(X509Certificate2Collection certificates)
+        {
+            if (certificates == null || certificates.Count == 0)
+            {
+                if (Logging.IsEnabled) Logging.Info(null, $"{nameof(CertificateInstaller)} parameter 'certificates' was null or empty.");
```

Comment:
nit: use `nameof(certificates)` instead

Reply:
so should we just peacefully return here? no exceptions? do we expect this to be called with a null value?

Reply:
The caller ensures this is not called when null. However we can choose to throw an exception too but I wanted to keep the code same in DPS and DeviceClient so it is easy to maintain if we make changes. It would have been good to have it common but this file is already part of DPS so I just copied it here as well.

Reply:
so with that logic this whole `if` statement is unnecessary. 
I would argue that this class should be self contained. I think an exception should be thrown here. 
If we want to reuse the same code, we can keep it in shared.

Reply:
But we cannot move it out of our existing DPS package so we will have two copies anyway. I am in for updating this file here though. I am inclined towards not having the if statement here. This is a helper so all it should do is install if there is something in the list. The caller should take care of exception of not calling when null.